### PR TITLE
#3812 Paper Connector action with Connector api (create, start, stop, remov…

### PR DESCRIPTION
**Pull request by wu87988622**
…e, update)

### Checklist
- [x] [Review Contribution Guidelines](https://ohara.readthedocs.io/en/latest/contributing.html)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#ohara-team))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #3812 )
- [ ] Add unit tests (or please explain why this PR is not worth having new tests)
- [ ] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: ohara-manager/client/src/components/Pipeline/Pipeline.js [hunk 1]
```diff
@@ -114,6 +127,38 @@ const Pipeline = () => {
...
+                      case KIND.topic:
+                        await createTopic({ ...cell, ref });
+                        break;
+
+                      default:
+                        break;
+                    }
+                  }}
+                  onCellStart={(id, name, paperApi) => {
+                    startConnector({ id, name, ref: paperApi });
```
Comment (eechih) [inline ohara-manager/client/src/components/Pipeline/Pipeline.js]: ref 的名稱太模糊了，有識別度的名稱比較好，例如：paper。 :)

File: ohara-manager/client/src/components/Pipeline/Toolbox/toolboxUtils.js [hunk 2]
```diff
@@ -237,6 +243,7 @@ export const enableDragAndDrop = params => {
             displayName: name,
             classType,
             className,
+            jarKey: jarKey && jarKey,
```
Comment (eechih) [inline ohara-manager/client/src/components/Pipeline/Toolbox/toolboxUtils.js]: jarKey && jarKey 😲typo?

File: ohara-manager/client/src/components/Pipeline/Toolbox/ToolboxHooks.js [hunk 3]
```diff
@@ -54,23 +54,33 @@ export const useFiles = () => {
...
       if (streamClasses.length > 0) {
         const results = streamClasses
-          .map(({ className, classType }) => {
-            const name = className.split('.').pop();
-            return {
-              name,
-              classType,
-              className,
-            };
+          .map(streamClasse => {
```
Comment (eechih) [inline ohara-manager/client/src/components/Pipeline/Toolbox/ToolboxHooks.js]: Typo: streamClasse => streamClass

File: ohara-manager/client/src/components/Pipeline/Toolbox/ToolboxHooks.js [hunk 4]
```diff
@@ -54,23 +54,33 @@ export const useFiles = () => {
...
+                jarKey: {
+                  name: streamClasse.name,
+                  group: streamClasse.group,
+                },
+              };
+            });
           })
-          .sort((a, b) => a.className.localeCompare(b.className));
+          .sort((a, b) => a.className.localeCompare(b.className))
+          .flat();
```
Comment (eechih) [inline ohara-manager/client/src/components/Pipeline/Toolbox/ToolboxHooks.js]: 👍

File: ohara-manager/client/src/components/Pipeline/Cells/ConnectorCell.js [hunk 5]
```diff
@@ -142,23 +143,23 @@ const ConnectorCell = options => {
       });
 
       // Start
-      $startButton.on('click', () => {
-        if (_.isFunction(onCellStart)) onCellStart(id, name);
+      $startButton.on('click', async () => {
+        if (_.isFunction(onCellStart)) await onCellStart(id, name, paperApi);
```
Comment (eechih) [inline ohara-manager/client/src/components/Pipeline/Cells/ConnectorCell.js]: 加上 async, await 有何用意？👆以這個 Case 為例，我認為在 onCellStart(...) 之後，並沒有任何事情等待處理，就不用特別加上 async/await。

Comment (wu87988622) [reply ohara-manager/client/src/components/Pipeline/Cells/ConnectorCell.js]: 我必須等待start 跑完再去更新paper.

File: ohara-manager/client/src/components/Pipeline/Pipeline.js (hunk 1 above)
Comment (wu87988622) [reply ohara-manager/client/src/components/Pipeline/Pipeline.js]: ok nice

File: ohara-manager/client/src/components/Pipeline/Toolbox/toolboxUtils.js (hunk 2 above)
Comment (wu87988622) [reply ohara-manager/client/src/components/Pipeline/Toolbox/toolboxUtils.js]: yes~ thanks

File: ohara-manager/client/src/components/Pipeline/Toolbox/ToolboxHooks.js (hunk 3 above)
Comment (wu87988622) [reply ohara-manager/client/src/components/Pipeline/Toolbox/ToolboxHooks.js]: thanks

**Pull request merged**